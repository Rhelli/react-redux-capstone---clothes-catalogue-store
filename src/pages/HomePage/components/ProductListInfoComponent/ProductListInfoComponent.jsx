import React from 'react';
import PropTypes from 'prop-types';
import styles from './ProductListInfoComponent.module.scss';

const ProductListInfoComponent = ({ productName, price }) => (
  <div className={styles.productListInfoContainer}>
    <div className={styles.productListInfoNameContainer}>
      <h3>{productName}</h3>
    </div>
    <div className={styles.productListInfoPriceContainer}>
      <h5>
        Â£
        {price}
      </h5>
    </div>
  </div>
);

ProductListInfoComponent.propTypes = {
  productName: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};

export default ProductListInfoComponent;
