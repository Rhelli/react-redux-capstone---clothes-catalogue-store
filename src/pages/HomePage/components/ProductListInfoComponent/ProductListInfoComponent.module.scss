.productListInfoContainer {
  width: 100%;
  margin-bottom: 15px;
  padding: 0 10px;
}

.productListInfoNameContainer {
  h3 {
    padding: 4px 0;
    font: 400 16px/16px 'Playfair Display', monospace;
  }
}

.productListInfoPriceContainer {
  h5 {
    font: 400 12px/12px 'Source Sans Pro', monospace;
  }
}

@media only screen and (min-width: 768px) {
  .productListInfoNameContainer {
    h3 {
      font-size: 28px;
      line-height: 32px;
    }
  }

  .productListInfoPriceContainer {
    h5 {
      font-size: 20px;
      line-height: 20px;
    }
  }
}

@media only screen and (min-width: 1024px) {
  .productListInfoNameContainer {
    h3 {
      font-size: 24px;
      line-height: 30px;
    }
  }

  .productListInfoPriceContainer {
    margin-top: 10px;

    h5 {
      font-size: 18px;
      line-height: 18px;
    }
  }
}

@media only screen and (min-width: 1200px) {
  .productListInfoNameContainer {
    h3 {
      font-size: 22px;
      line-height: 24px;
    }
  }

  .productListInfoPriceContainer {
    margin-top: 10px;

    h5 {
      font-size: 18px;
      line-height: 18px;
    }
  }
}
