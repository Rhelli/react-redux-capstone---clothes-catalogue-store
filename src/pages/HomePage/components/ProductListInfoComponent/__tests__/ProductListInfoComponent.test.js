/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProductListInfoComponent from '../ProductListInfoComponent';

configure({ adapter: new Adapter() });

const productName = 'Black T-Shirt';
const price = '10.00';
const productListInfoComponent = <ProductListInfoComponent productName={productName} price={price} />;

describe('The appearance of the ProductListInfoComponent', () => {
  it('Matches the ProductListInfoComponent snapshot', () => {
    const tree = renderer.create(productListInfoComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has an outer container with the correct className', () => {
    const { container } = render(productListInfoComponent);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('productListInfoContainer')).toBe(true);
  });

  it('Has two child containers, both with the correct classNames', () => {
    const { container } = render(productListInfoComponent);
    expect(container.firstChild.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.firstChild.classList.contains('productListInfoNameContainer')).toBe(true);
    expect(container.firstChild.lastChild.nodeName).toBe('DIV');
    expect(container.firstChild.lastChild.classList.contains('productListInfoPriceContainer')).toBe(true);
  });

  it('Displays both the products name as well as the products price, within the correct tags', () => {
    const { container, getByText } = render(productListInfoComponent);
    expect(container.firstChild.firstChild.firstChild.nodeName).toBe('H3');
    getByText('Black T-Shirt');
    expect(container.firstChild.lastChild.firstChild.nodeName).toBe('H5');
    getByText('Â£10.00');
  });
});
