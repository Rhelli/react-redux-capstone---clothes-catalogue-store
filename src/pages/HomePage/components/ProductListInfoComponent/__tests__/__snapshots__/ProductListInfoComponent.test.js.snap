// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the ProductListInfoComponent Matches the ProductListInfoComponent snapshot 1`] = `
<div
  className="productListInfoContainer"
>
  <div
    className="productListInfoNameContainer"
  >
    <h3>
      Black T-Shirt
    </h3>
  </div>
  <div
    className="productListInfoPriceContainer"
  >
    <h5>
      Â£
      10.00
    </h5>
  </div>
</div>
`;
